GomokuServer/src/GomokuServer.Api/Extensions/ArdalisResultExtensions.cs

```diff
@@ -43,4 +45,34 @@ public static IActionResult ToApiResponse(this Result result)
 
 		return new NoContentResult();
 	}
+
+	public static ErrorMessage GetHubError<T>(this Result<T> result)
+	{
+		if (result.IsNotFound())
+		{
+			return new ErrorMessage(string.Join(", ", result.Errors));
+		}
+
+		if (result.IsInvalid())
+		{
+			return new ErrorMessage(string.Join(", ", result.ValidationErrors.Select(error => error.ErrorMessage)));
+		}
+
+		return new ErrorMessage("Unexpected error occured");
+	}
+
+	public static ErrorMessage GetHubError(this Result result)
+	{
+		if (result.IsNotFound())
+		{
+			return new ErrorMessage(string.Join(", ", result.Errors));
+		}
+
+		if (result.IsInvalid())
+		{
+			return new ErrorMessage(string.Join(", ", result.ValidationErrors.Select(error => error.ErrorMessage)));
+		}
+
+		return new ErrorMessage("Unexpected error occured");
+	}
```

Comment:
I suggest we start considering more explicit error messages that clearly state what went wrong, as this will save us time in the future. Could you please update this ErrorMessage to be more descriptive? Alternatively, it would be even better to create a separate error logging class that can be used as a wrapper for passing detailed messages, etc

Reply:
This last line is just fallback in case if some network request errors occured or stuff like that. Usually, If something from logic side is incorrect, you will receive Validation Errors:

![image](https://github.com/user-attachments/assets/d29d9233-bf03-43b6-b760-f6d6685ec55c)

Reply:
I see, thanks for clarification, but don't we want to follow here little bit more DRY, as I do see here some duplication stuff. As you can have shared logic between this methods

```c#
public static class ResultExtensions
{
    public static ErrorMessage GetHubError(this IResult result)
    {
        if (result.IsNotFound())
        {
            return new ErrorMessage(string.Join(", ", result.Errors));
        }

        if (result.IsInvalid())
        {
            return new ErrorMessage(string.Join(", ", result.ValidationErrors.Select(error => error.ErrorMessage)));
        }

        return new ErrorMessage("Unexpected error occurred");
    }
}

public class Result<T> : IResult
{
    public bool IsNotFound() { /*TODO implementation */ }
    public bool IsInvalid() { /* implementation */ }
    public IEnumerable<string> Errors { get; set; }
    public IEnumerable<ValidationError> ValidationErrors { get; set; }
}

public class Result : IResult
{
    public bool IsNotFound() { /*TODO implementation */ }
    public bool IsInvalid() { /* implementation */ }
    public IEnumerable<string> Errors { get; set; }
    public IEnumerable<ValidationError> ValidationErrors { get; set; }
}

....
```

Reply:
Good point. Applied suggested changes